### PR TITLE
Update Documentation: SelectControl readme

### PR DESCRIPTION
**Pull request by melchoyce**
Initial commit to update SelectControl readme. These changes add design documentation.

## Description

This PR adds design guideline documentation in addition to the development documentation that existed previously. These guidelines are 'best practices' for the usage of the component, as well as describing the component in more detail.

## Screenshots

A preview can be seen here, although it doesn't represent what it will look like in the handbook:

http://simp.ly/publish/3pZlzc

## Feedback

I'd appreciate feedback on:

- Whether my markdown syntax is correct
- Design guidelines
- Finding a way to preview these changes and see what they'll look like in the handbook

Thank you!

### CODE REVIEW

**Comment by chrisvanpatten** [comment]
There are a few small markdown things I'll put in a review but overall this looks great!

One thing I want to get @mtias's thoughts on are the images. I know at one point he had mentioned it wouldn't be ideal to have documentation images inline in the repo, and I agree (it adds up to a big repo download size) but we'll need to figure out a better approach.

Some projects just upload the images to GitHub issues and then copy out the URLs and hope GitHub doesn't change them… I'm not sure that's a great approach. Do any of the .org projects face this problem? How might we solve this?

**Comment by melchoyce** [comment]
We host the About page images in the WordPress CDN — maybe that's the right place for them here, too? (@pento knows more about this)

**Comment by mtias** [comment]
Uploading to WordPress.org would be a bit better in my view.

**Comment by chrisvanpatten** [comment]
That makes sense to me. Should I ping someone in #meta, or can @pento help coordinate that?

**Comment by melchoyce** [comment]
Let's upload them to https://wordpress.org/gutenberg/ — I have access to that, so I can upload my own images. We can coordinate images for other PRs in Slack, maybe.

@chrisvanpatten Did you have some markdown feedback?

**Review by chrisvanpatten — DISMISSED** [review]
@melchoyce Sorry for the mixup on this and #12485 :) The same general markdown formatting notes from #12485 apply here. I'm happy to push up changes to the branch for you if you want, or makes specific review notes, or something else. Let me know what's best for you :)

**Comment by melchoyce** [comment]
Pushed up some changes. @chrisvanpatten, can I get you to review?

File: packages/components/src/select-control/README.md [hunk 1]
```diff
@@ -1,89 +1,180 @@
 # SelectControl
 
-SelectControl component is used to generate select input fields.
+SelectControl allow users to select from a single-option menu. It functions as a wrapper around the browser's native `<select>` element.
 
 
-## Usage
+![A “Link To” select with “none” selected.](https://wordpress.org/gutenberg/files/2018/12/select.png)
+
+
```
Comment (kjellr) [inline packages/components/src/select-control/README.md]: I see you removed some extra line breaks, but I still see some here and around the images. Other than that, this is looking good. 👍

**Review by mapk — APPROVED** [review]
My time reviewing one of these, so I'm not exactly sure what I should be looking for. I checked for spelling, and readability. And made sure images aligned with descriptions. Please school me in slack if I should be checking for more. Otherwise, it all looks good.

File: packages/components/src/select-control/README.md (hunk 1 above)
Comment (melchoyce) [reply packages/components/src/select-control/README.md]: Thanks, updated!

**Review by kjellr — APPROVED** [review]
🚢 LGTM!

**Comment by kjellr** [comment]
> @melchoyce Sorry for the mixup on this and #12485 :) The same general markdown formatting notes from #12485 apply here. I'm happy to push up changes to the branch for you if you want, or makes specific review notes, or something else. Let me know what's best for you :)

@chrisvanpatten I think these are all set, so I'm going to dismiss this review for now. 👍

**Pull request merged**